org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/LRUCache.java

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.jdt.ls.debug.adapter;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
+public class LRUCache<K, V> extends LinkedHashMap<K, V> {
+    private static final long serialVersionUID = -7068164191168103891L;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
```

Comment:
>DEFAULT_LOAD_FACTOR [](start = 31, length = 19)

correct me if i was misundertanding it .
is there any way to disable the auto-augmenting the cache? it seems useless as we would never exceed the cachesize since  you overwrite the removeEldestEntry method?

Reply:
i think makes the load factor 1.0 would disable it

---
In reply to: [136267793](https://github.com/vscjavaci/eclipse.jdt.ls/pull/75#discussion_r136267793) [](ancestors = 136267793)

Reply:
the map will increase the inner table if the size of the map has hit the current capacity * load factor, I will make a initial capacity of cache_size / load factor + 1 to avoid re-alloc of the map.

Reply:
got it. i miss the point that its inner implementation is hashmap